### DISCUSSION TITLE
Strict mode disabled but development still has extra rerenders

### DISCUSSION THREAD

**Discussion by kmcaloon** [vercel/next.js#53575] [Help]
### Summary

So when in development we are seeing a provider component rerender 4 times on initial page load, whereas in production it correctly only renders once. Strict mode is disabled, and I can see when we enable it we see extra renders in addition to the 4.

I'm continuing to dig in the codebase, but I was wondering if development has something in place even when strict mode is disabled that would be causing this?

**Comment by icyJoseph** [comment]
Is it using pages router? that stack has extra renders when there's queries.

- https://nextjs.org/docs/pages/building-your-application/rendering/automatic-static-optimization#how-it-works

> During prerendering, the router's query object will be empty since we do not have query information to provide during this phase. After hydration, Next.js will trigger an update to your application to provide the route parameters in the query object.

It could be one reason, but Idk if your app is doing something extra.

**Reply by kmcaloon** [reply]
Thanks for the reply. So I'm actually using the app directory. I'll play around with it though.
